File: Python/Program/Motorcycle.py
'''
Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 2 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
'''

from Vehicle import Vehicle

class Motorcycle(Vehicle):
    # Private Attributes
    __jenisMotor = ""
    __kapasitasTangki = ""
    
    def __init__(self, platNomor="", merk="", tahunProduksi="", warna="", jenisMotor="", kapasitasTangki=""):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        self.__jenisMotor = jenisMotor
        self.__kapasitasTangki = kapasitasTangki
        
    # Getter and Setter
    # Getter 
    def getJenisMotor(self):
        return self.__jenisMotor

    def getKapasitasTangki(self):
        return self.__kapasitasTangki
    
    # Setter 
    def setJenisMotor(self, jenisMotor):
        self.__jenisMotor = jenisMotor

    def setKapasitasTangki(self, kapasitasTangki):
        self.__kapasitasTangki = kapasitasTangki

File: Cpp/Program/Motorcycle.cpp
/*Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 4 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Motorcycle : public Vehicle 
{
    /*Attribute*/
    private:
        string jenisMotor;

    public:
        /*Constructor*/
        Motorcycle(){

        }

        Motorcycle(string platNomor, string merk, string tahunProduksi, string warna, string kapasitasTangki, string jenisMotor) : Vehicle(platNomor, merk, tahunProduksi, warna, kapasitasTangki) {
            this->jenisMotor = jenisMotor;
        }

        /*Getter & Setter*/
        //Getter
        string getJenisMotor() {
            return this ->jenisMotor;
        }

        //Setter
        void setJenisMotor(string jenisMotor) {
            this->jenisMotor = jenisMotor;
        }

        ~Motorcycle()
        {}
};
